File: frontend/src/types/product.ts
export interface Product {
  id: number;
  business_id: number;
  name: string;
  description?: string;
  price: number;
  category?: string;
  tags?: string;
  stock: number;
  low_stock_threshold: number;
  image_url?: string;
  is_active: boolean;
  is_featured: boolean;
  is_low_stock: boolean;
  created_at: string;
  updated_at?: string;
}

export interface ProductCreate {
  name: string;
  description?: string;
  price: number;
  category?: string;
  tags?: string;
  stock?: number;
  low_stock_threshold?: number;
  image_url?: string;
  is_featured?: boolean;
}

export interface ProductUpdate extends Partial<ProductCreate> {
  is_active?: boolean;
}

File: backend/app/schemas/product.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from decimal import Decimal


class ProductBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    price: Decimal = Field(..., gt=0)
    category: Optional[str] = None
    tags: Optional[str] = None
    stock: int = Field(default=0, ge=0)
    low_stock_threshold: int = Field(default=5, ge=0)
    image_url: Optional[str] = None
    is_featured: Optional[bool] = False


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    price: Optional[Decimal] = None
    category: Optional[str] = None
    tags: Optional[str] = None
    stock: Optional[int] = None
    low_stock_threshold: Optional[int] = None
    image_url: Optional[str] = None
    is_active: Optional[bool] = None
    is_featured: Optional[bool] = None


class Product(ProductBase):
    id: int
    business_id: int
    is_active: bool
    is_low_stock: bool = False
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True
